internal/controllers/machine/machine_controller_test.go

```diff
@@ -2874,11 +2923,13 @@ func TestIsDeleteNodeAllowed(t *testing.T) {
 				mcpBeingDeleted,
 				empBeingDeleted,
 			).Build()
+			remoteClient := fake.NewClientBuilder().WithIndex(&corev1.Node{}, "spec.providerID", index.NodeByProviderID).WithObjects(testNodeA).Build()
```

Comment:
How about creating remote client out side of subtest ?
It seems to be good to create it just once.